.github/scripts/generate_linux_ci_workflows.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+import jinja2
+
+import os.path
+from pathlib import Path
+
+DOCKER_REGISTRY = "308535385114.dkr.ecr.us-east-1.amazonaws.com"
+
+GITHUB_DIR = Path(__file__).parent.parent
+
+
+class PyTorchLinuxWorkflow:
+    def __init__(self, build_environment: str, docker_image_base: str):
+        self.build_environment = build_environment
+        self.docker_image_base = docker_image_base
+
+    def generate_workflow_file(
+        self, workflow_template: jinja2.Template, jinja_env: jinja2.Environment
+    ) -> Path:
+        output_file_path = GITHUB_DIR.joinpath(
+            f"workflows/{self.build_environment}.yml"
+        )
+        with open(output_file_path, "w") as output_file:
+            output_file.write(
+                workflow_template.render(
+                    build_environment=self.build_environment,
+                    docker_image_base=self.docker_image_base,
+                )
+            )
```

Comment:
To preserve the trailing newline (copied from https://github.com/pytorch/pytorch/pull/54344/commits/2807422cae1f95aa9ea7950027f9d45b3adc5b13):
```suggestion
            )
            output_file.write('\n')
```